### PR TITLE
Update 2022-11-19-monitoring-with-uncertainty.md

### PR DESCRIPTION
**Pull request by cmougan**
Python code snippet

### CODE REVIEW

**Comment by cmougan** [comment]
Added a snippet of code and made some minimal changes :)

**Comment by cmougan** [comment]
@saattrupdan perhaps in the preprint, in the comment, you can add: "AAAI'23 Safe and Robust AI track"

**Comment by saattrupdan** [comment]
> @saattrupdan perhaps in the preprint, in the comment, you can add: "AAAI'23 Safe and Robust AI track"

Updated now - it'll take ~1-2 days for ArXiv to update, however.

File: _posts/2022-11-19-monitoring-with-uncertainty.md [hunk 1]
```diff
@@ -145,7 +146,28 @@ Of course, this does not _directly_ show the reason for the deterioration, but i
...
+y = x1 ** 2 + x2 + np.random.normal(0, 0.01, 10000)
+clf = Boot(LinearRegression())
+clf = clf.fit(X, y)
+
+preds, intervals = clf.predict(X_ood, uncertainty=0.05)
+unc = intervals[:, 1] - intervals[:, 0]
+
+m = LinearRegression().fit(X_ood, unc)
+np.round(m.coef_, decimals=2)
+#[ 0.01,  0.  , -0.  ]
```
Comment (saattrupdan) [inline _posts/2022-11-19-monitoring-with-uncertainty.md:169]: Adding comments:

```suggestion
# Generate normal-distributed random data
x1 = np.random.normal(1, 0.1, size=10000)
x2 = np.random.normal(1, 0.1, size=10000)
x3 = np.random.normal(1, 0.1, size=10000)

# Create a synthetic dataset with the random data, of shape (1, 3)
X = np.array([x1, x2, x3]).T

# Create out-of-distribution data by shifting the first feature by 5
X_ood = np.array([x1 + 5, x2, x3]).T

# Create the target variable, which depends non-linearly on `x1`, linearly on `x2`, and does not depend on `x3` at all
y = x1 ** 2 + x2 + np.random.normal(0, 0.01, 10000)

# Create linear regression model with uncertainty estimation support, using our `Boot` wrapper class
clf = Boot(LinearRegression())

# Fit the model to the data
clf = clf.fit(X, y)

# Compute predictions along with prediction intervals on the out-of-distribution data
preds, intervals = clf.predict(X_ood, uncertainty=0.05)

# Compute the uncertainty, being the width of the prediction intervals
unc = intervals[:, 1] - intervals[:, 0]

# As for explaining where the uncertainty comes from, we fit a new linear regression model
# on the out-of-distribution data, which attempts to predict the uncertainties
m = LinearRegression().fit(X_ood, unc)

# Print out the coefficients of the second model, which corresponds to the SHAP values.
# We see that it puts no importance on any of the variables, as they are merely random
np.round(m.coef_, decimals=2)
#[ 0.01,  0.  , -0.  ]
```

**Review by saattrupdan — CHANGES_REQUESTED** [review]

**Review by cmougan — COMMENTED** [review]
Good idea to comment

**Review by saattrupdan — APPROVED** [review]
LGTM!

**Pull request merged**